sdjournal/journal.go

```diff
@@ -266,12 +549,17 @@ func (j *Journal) GetDataValue(field string) (string, error) {
 // turned off by setting it to 0, so that the library always returns the
 // complete data objects.
 func (j *Journal) SetDataThreshold(threshold uint64) error {
+	sd_journal_set_data_threshold, err := j.getFunction("sd_journal_set_data_threshold")
+	if err != nil {
+		return err
+	}
+
 	j.mu.Lock()
-	r := C.sd_journal_set_data_threshold(j.cjournal, C.size_t(threshold))
+	r := C.my_sd_journal_set_data_threshold(sd_journal_set_data_threshold, j.cjournal, C.size_t(threshold))
```

Comment:
Yes, the (hypothetical) problem is the implicit conversion from incoming `uint64` to 32-bits `size_t`. Anything bigger than `UINT32_MAX` could result in unexpected values (depending on what golang does in this case).
As this was here before and it isn't a major issue, let's split this discussion out of this PR review so this can proceed.